com.reprezen.swagedit/src/com/reprezen/swagedit/json/JsonUtil.java

```diff
@@ -49,21 +42,52 @@ public static SchemaDefinition getReference(JsonNode document, JsonNode refNode)
 			ref = ref.substring(ref.indexOf("#"));
 		}
 
-		final String[] keys = (ref.startsWith("#/") ? ref.substring(2) : ref).split("/");
+		JsonPointer pointer = asPointer(ref);
+		JsonNode found = document.at(pointer);
+		String ptr = pointer.toString();
+		String description = ptr.substring(
+				ptr.lastIndexOf("/") + 1, 
+				ptr.length());
 
-		found = document;
-		String lastKey = null;
-		for (String key : keys) {
-			JsonNode value = found.get(key);
-			if (value != null) {
-				found = value;
-				lastKey = key;
-			}
+		return new SchemaDefinition(document, !found.isMissingNode() ? found : refNode, description);
+	}
+
+	public static boolean isPointer(String ptr) {
+		String sanitized = sanitize(ptr);
+		if (sanitized == null) {
+			return false;
 		}
+		return sanitized.startsWith("#");
+	}
+
+	public static JsonPointer asPointer(String ptr) {
+		String sanitized = sanitize(ptr);
+		if (sanitized == null) {
+			return JsonPointer.compile("");
+		}
+		if (sanitized.startsWith("#")) {
+			sanitized = sanitized.substring(1);
+		}
+		return JsonPointer.compile(sanitized);
+	}
 
-		return new SchemaDefinition(document, found != null ? found : refNode, lastKey);
+	public static JsonPointer getPointer(JsonNode ref) {
```

Comment:
`getPointer` and `asPointer` share the following code:
```java
		if (asText.startsWith("#")) {
			asText = asText.substring(1);
		}

		return JsonPointer.compile(asText);
```
It can be reused